demos/shapes-custom-question-authoring-demo/src/question/index.js

```diff
@@ -0,0 +1,339 @@
+import { PREFIX, mapSizeStringToFactor } from './constants';
+
+export default class Question {
+    constructor(init, lrnUtils) {
+        this.init = init;
+        this.events = init.events;
+        this.lrnUtils = lrnUtils;
+        this.el = init.$el.get(0);
+
+        this.render().then(() =>{
+            this.registerPublicMethods();
+            this.handleEvents();
+
+            /**
+             * @param { String } init.state - the state of Questions API.
+             * state can be any of the following 3 strings
+             * "initial" for first starting the assessment,
+             * "resume" for coming back to a previously started assessment,
+             * "review" for showing the completed assessment and results to the learner or teacher
+             */
+             
+             if (init.state === 'resume') {
+                /**
+                 * If you want your custom question to support resume state
+                 * (For a learner to be able to come back to your question in a future sitting after having previously started and saved the assessment),
+                 * then make make sure to update your question's UI to display the previously saved response Questions API has stored in the back end.
+                 */
+
+                if(init.response) {
+                    const shapeButtons = Array.from(this.el.querySelectorAll('.shape-wrapper button'));
+                    shapeButtons.forEach((shapeButton) => {
+                        if(shapeButton.getAttribute('data-index') === init.response) {
+                            shapeButton.classList.add('selected');
+                        }
+                    })
+                }
+            }
```

Comment:
I think this is looks like a duplicate of line:46 , maybe we can create one function that can be shared for both 
```suggestion
                if (init.response) {
                    initResponse();
            }
```